website/docs/r/compute_disk.html.markdown

```diff
@@ -1,102 +1,231 @@
 ---
+# ----------------------------------------------------------------------------
+#
+#     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+#
+# ----------------------------------------------------------------------------
+#
+#     This file is automatically generated by Magic Modules and manual
+#     changes will be clobbered when the file is regenerated.
+#
+#     Please read more about how to change this file in
+#     .github/CONTRIBUTING.md.
+#
+# ----------------------------------------------------------------------------
 layout: "google"
 page_title: "Google: google_compute_disk"
 sidebar_current: "docs-google-compute-disk"
 description: |-
-  Creates a new persistent disk within GCE, based on another disk.
+  Persistent disks are durable storage devices that function similarly to
+  the physical disks in a desktop or a server.
 ---
 
 # google\_compute\_disk
 
-Creates a new persistent disk within GCE, based on another disk. For more information see
-[the official documentation](https://cloud.google.com/compute/docs/disks/add-persistent-disk)
-and
-[API](https://cloud.google.com/compute/docs/reference/latest/disks).
-
-~> **Note:** All arguments including the disk encryption key will be stored in the raw state as plain-text.
-[Read more about sensitive data in state](/docs/state/sensitive-data.html).
-
-## Example Usage
-
-```hcl
-resource "google_compute_disk" "default" {
-  name  = "test-disk"
-  type  = "pd-ssd"
-  zone  = "us-central1-a"
-  image = "debian-8-jessie-v20170523"
-  labels {
-    environment = "dev"
-  }
-}
-```
-
-## Argument Reference
+Persistent disks are durable storage devices that function similarly to
+the physical disks in a desktop or a server. Compute Engine manages the
+hardware behind these devices to ensure data redundancy and optimize
+performance for you. Persistent disks are available as either standard
+hard disk drives (HDD) or solid-state drives (SSD).
 
-The following arguments are supported:
+Persistent disks are located independently from your virtual machine
+instances, so you can detach or move persistent disks to keep your data
+even after you delete your instances. Persistent disk performance scales
+automatically with size, so you can resize your existing persistent disks
+or add more persistent disks to an instance to meet your performance and
+storage space requirements.
 
-* `name` - (Required) A unique name for the resource, required by GCE.
-    Changing this forces a new resource to be created.
-
-* `zone` - (Required) The zone where this disk will be available.
-
-- - -
+Add a persistent disk to your instance when you need reliable and
+affordable storage with consistent performance characteristics.
```

Comment:
Can you add a `ReferenceLinks` section to api.yaml so we don't lose links to the docs?

Reply:
Done.